src/moduls/render.ts

```diff
@@ -0,0 +1,2146 @@
+            stepleft.innerHTML = '0';
+            stepright.innerHTML = '15';
+            stepright1.innerHTML = '45';
+            stepleft1.innerHTML = '10';
+            Object(searchValue).value = '';
+            this.renderCards(flowers);
+        }
+    }
+
+    sort(e: MouseEvent) {
+        if (Object(e).target.classList.contains('sortab')) {
+            sortab.classList.toggle('active');
+            sortba.classList.remove('active');
+            const sortArr = cardsD.sort((a, b) => (b.name > a.name ? 1 : -1));
+
+            this.renderCardsD(sortArr);
+        }
+
+        if (Object(e).target.classList.contains('sort12')) {
+            sort12.classList.toggle('active');
+            sort21.classList.remove('active');
+            const sortArr = cardsD.sort((a, b) => (b.much > a.much ? 1 : -1));
+
+            this.renderCardsD(sortArr);
+        }
+
+        if (Object(e).target.classList.contains('sortba')) {
+            sortba.classList.toggle('active');
+            sortab.classList.remove('active');
+            const sortArr = cardsD.sort((a, b) => (a.name > b.name ? 1 : -1));
+
+            this.renderCardsD(sortArr);
+        }
+        if (Object(e).target.classList.contains('sort21')) {
+            sort21.classList.toggle('active');
+            sort12.classList.remove('active');
+            const sortArr = cardsD.sort((a, b) => (a.much > b.much ? 1 : -1));
+
+            this.renderCardsD(sortArr);
+        }
+    }
+
+    diapozon() {
+        if (Object(+sliders[1]).value < Object(+sliders[0]).value) {
+            Object(sliders[0]).value = Object(+sliders[1]).value;
+        }
+
+        if (Object(+sliders[0]).value > Object(+sliders[1]).value) {
+            Object(sliders[1]).value = Object(+sliders[0]).value;
+        }
+
+        if (Object(+sliders[3]).value < Object(+sliders[2]).value) {
+            Object(sliders[2]).value = Object(+sliders[3]).value;
+        }
+
+        if (Object(+sliders[2]).value > Object(+sliders[3]).value) {
+            Object(sliders[3]).value = Object(+sliders[2]).value;
+        }
+
+        sliders.forEach((slider) => {
+            slider.addEventListener('change', () => {
+                stepleft.innerHTML = Object(sliders[0]).value;
+                stepright.innerHTML = Object(sliders[1]).value;
+
+                const diap = cardsD.filter(
+                    (key) =>
+                        Object(sliders[0]).value <= Object(key).much / 1 &&
+                        Object(key).much / 1 <= Object(sliders[1]).value
+                );
+
+                stepleft1.innerHTML = Object(sliders[2]).value;
+                stepright1.innerHTML = Object(sliders[3]).value;
+
+                const flength = diap.filter(
+                    (key) => Object(sliders[2]).value <= key.flength && key.flength <= Object(sliders[3]).value
+                );
+                this.renderCardsD(flength);
+            });
+        });
+    }
+
+    filter(e: MouseEvent) {
```

Comment:
It would be easier to split the logic into steps. The implemenation might be as the following:
1. Get values for all filters (e.g. active or not)
2. Get filters into some data structure (e.g. an array with selected colors, array with selected product types, ranges, etc.)
3. Apply all filters one by one to a list of cards - e.g. by color first, then by size, then by type.
4. Render cards that satisfy all filters.
5. Get elements for filters and change their status - e.g. if red color is selected, set class for the red. This can be automated using additional data or using closures or just by naming convention.